File: 3638-maximum-balanced-shipments/3638-maximum-balanced-shipments.java
class Solution {
    public int maxBalancedShipments(int[] weight) {
        int count = 0;
        int max = Integer.MIN_VALUE;

        for(int i=0; i<weight.length; i++){
            max = Math.max(max, weight[i]);

            if(weight[i] < max){
                count++;
                max = Integer.MIN_VALUE;
            }
        }

        return count;
    }
}

File: 3638-maximum-balanced-shipments/3638-maximum-balanced-shipments.cpp
class Solution {
public:
    int maxBalancedShipments(vector<int>& weight) {
        int count = 0;
        int mx = INT_MIN;

        for(int i=0; i<weight.size(); i++){
            mx = max(mx, weight[i]);

            if(weight[i] < mx){
                count++;
                mx = INT_MIN;
            }
        }

        return count;
    }
};